### PR TITLE
Add FAQ File

### PR DESCRIPTION
**Pull request by mawenjie8731**
### PR types
New features

### PR changes
 Docs

### Description
Add FAQ File

### CODE REVIEW

File: docs/FAQ .md [hunk 1]
```diff
@@ -0,0 +1,509 @@
+## FAQ
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 文件怎么多了个空格呢

File: docs/FAQ .md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+
+## 写在前面
+
++ 我们收集整理了开源以 来在Issues和用户群中的常见问题并且给出了简要解答，旨在为NLP的开发者提供一些参考，也希望帮助大家少走一些弯路。
++ NLP领域大佬众多，本文档回答主要依赖有限的项目实践，难免挂一漏万，如有遗漏和不足，也**希望有识之士帮忙补充和修正**，万分感谢。
++ 使用过程中可参考PaddleNLP[官方文档](https://paddlenlp.readthedocs.io/zh/latest/index.html) 、[AI Studio官方项目](https://aistudio.baidu.com/aistudio/projectdetail/1535371)。
+
+## PaddleNLP常见问题汇总（持续更新）
+
++ [【精选】NLP精选5问](#NLP精选)
```
Comment (chenxiaozeng) [inline docs/FAQ .md]: 这一级标题可以加下⭐️标识，突出一下子

File: docs/FAQ .md [hunk 3]
```diff
@@ -0,0 +1,509 @@
+## FAQ
+
+
+## 写在前面
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 删除这段。

File: docs/FAQ .md [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+
+```python
+from paddlenlp.datasets import load_dataset
+
+def read(data_path):
+    with open(data_path, 'r', encoding='utf-8') as f:
+        # 跳过列名
+        next(f)
+        for line in f:
+            words, labels = line.strip("\n').split("\t')
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 标点符号双引号映射不准确。一个双引号一个单引号

File: docs/FAQ .md [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+            words = words.split("\002')
+            labels = labels.split("\002')
+            yield {'tokens': words, 'labels': labels}
+
+# data_path为read()方法的参数
+map_ds = load_dataset(read, data_path='train.txt',lazy=False)
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 代码风格不准确，lazy前面要有空格

File: docs/FAQ .md [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+iter_ds = load_dataset(read, data_path='train.txt',lazy=True)
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 同上

File: docs/FAQ .md [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+
+**A:** 增加训练样本带来的效果是最直接的。此外，可以基于我们开源的[预训练模型](https://github.com/PaddlePaddle/PaddleNLP/tree/develop/paddlenlp/transformers)进行热启，再用少量数据集fine-tune模型。此外，针对分类、匹配等场景，[小样本学习](https://github.com/PaddlePaddle/PaddleNLP/tree/develop/examples/few_shot)也能够带来不错的效果。
+
+<a name="1-5"></a>
+
+##### Q1.5 如何提升模型的性能，提升QPS？
+
+**A:** 从工程角度，对于服务器端部署可以使用[Paddle Inference](https://www.paddlepaddle.org.cn/documentation/docs/zh/guides/05_inference_deployment/inference/inference_cn.html)高性能预测引擎进行预测部署。对于Transformer类模型的GPU预测还可以使用PaddleNLP中提供的[FasterTransformer](https://github.com/PaddlePaddle/PaddleNLP/tree/develop/paddlenlp/ops)功能来进行快速预测，其集成了[NV FasterTransformer](https://github.com/NVIDIA/FasterTransformer)并进行了功能增强。
+
+从模型策略角度，可以使用一些模型小型化技术来进行模型压缩，如模型蒸馏和裁剪，通过小模型来实现加速。PaddleNLP中集成了TinyERNIE、TinyBERT这样一些通用小模型供下游任务微调使用。另外PaddleNLP提供了[模型压缩示例](https://github.com/PaddlePaddle/PaddleNLP/tree/develop/examples/model_compression)，实现了DynaBERT、TinyBERT、MiniLM等方法策略，可以参考对自己的模型进行蒸馏压缩。
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 没有TinyERNIE这个模型，只有ERNIE-Tiny

File: docs/FAQ .md [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+      ...
+
+      model = Model(...)
+      model.eval()
+
+      ...
+
+```
+
+（2）第二种方法：以Ernie为例，将模型输出的 tensor 设置 `stop_gradient` 为 True。可以使用 `register_forward_post_hook` 按照如下的方式尝试：
```
Comment (ZeyuChen) [inline docs/FAQ .md]: Ernie  -> ERNIE

File: docs/FAQ .md [hunk 9]
```diff
@@ -0,0 +1,509 @@
...
+
+##### Q5.3 GPU版的Paddle虽然能在CPU上运行，但是必须要有GPU设备吗？
+
+**A:** 不支持 GPU 的设备只能安装 CPU 版本的 PaddlePaddle。 GPU 版本的 PaddlePaddle 如果想只在 CPU 上运行，可以通过 `export CUDA_VISIBLE_DEVICES=-1` 来设置。
+
+<a name="7-4"></a>
+
+##### Q5.4  如何指定用CPU还是GPU训练模型？
+
+**A:** 一般我们的训练脚本提供了，`--use_cuda,` 或者 `--device` 选项，用户可以通过这些选项选择需要使用的设备。
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 理论上我们没有再提供--use_cuda这个选项了。应该全部是--device才对

File: docs/FAQ .md [hunk 10]
```diff
@@ -0,0 +1,509 @@
...
+**A:** 其主要依赖于二次构造数据来进行finetune，同时要更新termtree信息。wordtag分为两个步骤：
+（1）通过BIOES体系进行分词；
+（2）将分词后的信息和TermTree进行匹配。
+	因此我们需要：
+（1）分词正确，这里可能依赖于wordtag的finetune数据，来让分词正确；
+（2）wordtag里面也需要把分词正确后term打上相应的知识信息。wordtag自定义TermTree的方式将在后续版本提供出来。
+
+可参考[issue](https://github.com/PaddlePaddle/PaddleNLP/issues/822)。
+
+<a name="使用咨询问题"></a>
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 为什么这些地方要用<a>, 而不是 标准的markdown语法？

File: docs/FAQ .md [hunk 11]
```diff
@@ -0,0 +1,509 @@
...
+（1）动转静
+
+   动转静，即将动态图的模型转为可用于部署的静态图模型。
+   动态图接口更加易用，python 风格的交互式编程体验，对于模型开发更为友好，而静态图相比于动态图在性能方面有更绝对的优势。因此动转静提供了这样的桥梁，同时兼顾开发成本和性能。
+   可以参考官方文档 [动态图转静态图文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/04_dygraph_to_static/index_cn.html)，使用 `paddle.jit.to_static` 完成动转静。
+   另外，在 PaddleNLP 我们也提供了导出静态图模型的例子，可以参考 [waybill_ie 模型导出](https://github.com/PaddlePaddle/PaddleNLP/tree/develop/examples/information_extraction/waybill_ie/#%E6%A8%A1%E5%9E%8B%E5%AF%BC%E5%87%BA)。
+
+（2）借助Paddle Inference部署
+
+   动转静之后保存下来的模型可以借助Paddle Inference完成高性能推理部署。Paddle Inference内置高性能的CPU/GPU Kernel、并结合细粒度OP横向纵向融合等策略、并集成 TensorRT 实现模型推理的性能提升。具体可以参考文档 [Paddle Inference 简介](https://paddleinference.paddlepaddle.org.cn/master/product_introduction/inference_intro.html)。
```
Comment (ZeyuChen) [inline docs/FAQ .md]: 并。。。。。并。。。。。。 修改下这列并行结构句法。

File: docs/FAQ .md (hunk 8 above)
Comment (ZeyuChen) [reply docs/FAQ .md]: ERNIE作为模型名称时统一大写，只有API里面才大写+小写

File: docs/FAQ .md (hunk 10 above)
Comment (mawenjie8731) [reply docs/FAQ .md]: markdown直接生成目录会把最head的大标题也作为目录，使用这个格式能自定义目录更美观一些

**Comment by yingyibiao** [comment]
需要在readthedocs上面增加FAQ部分
（我们需要将所有的文档都汇总到readthedocs，后续readthedocs就是所有文档的统一入口）

**Review by ZeyuChen — APPROVED** [review]

**Comment by ZeyuChen** [comment]
这个markdown先合入一波，然后同步修改readthedocs主页。
